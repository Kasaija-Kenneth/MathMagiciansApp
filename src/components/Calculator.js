import React from 'react';

class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="container">
        <div className="calculator">
          <input type="text" className="input-output" value={0} disabled />
          <div className="div-line1">
            <button type="button" className="ac-btn">AC</button>
            <button type="button" className="plus-minus-btn">&#177;</button>
            <button type="button" className="percent-btn">&#37;</button>
            <button type="button" className="divide-btn">&#247;</button>
          </div>
          <div className="div-line2">
            <button type="button" className="seven-btn">7</button>
            <button type="button" className="eight-btn">8</button>
            <button type="button" className="nine-btn">9</button>
            <button type="button" className="multiply-btn">&#215;</button>
          </div>
          <div className="div-line3">
            <button type="button" className="four-btn">4</button>
            <button type="button" className="five-btn">5</button>
            <button type="button" className="six-btn">6</button>
            <button type="button" className="subtract-btn">&#8722;</button>
          </div>
          <div className="div-line4">
            <button type="button" className="one-btn">1</button>
            <button type="button" className="two-btn">2</button>
            <button type="button" className="three-btn">3</button>
            <button type="button" className="plus-btn">&#43;</button>
          </div>
          <div className="div-line5">
            <button type="button" className="zero-btn">0</button>
            <button type="button" className="fullstop-btn">&#8901;</button>
            <button type="button" className="equals-btn">&#61;</button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
