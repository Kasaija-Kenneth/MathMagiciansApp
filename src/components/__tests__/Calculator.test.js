import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import Calculator from '../Calculator';

describe('Test render Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Simulation of user clicking button 8 twice', () => {
    render(<Calculator />);
    const buttonBtn = screen.getByText(/8/i);
    userEvent.click(buttonBtn);
    userEvent.click(buttonBtn);
    expect(screen.getByText(/88/i)).toBeInTheDocument();
  });
});
