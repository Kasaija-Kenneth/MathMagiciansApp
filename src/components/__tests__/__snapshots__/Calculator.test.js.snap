// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Calculator component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="calculator"
  >
    <output
      className="input-output"
    />
    <div
      className="div-line1"
    >
      <button
        className="ac-btn"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="plus-minus-btn"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        ±
      </button>
      <button
        className="percent-btn"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="divide-btn"
        name="÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      className="div-line2"
    >
      <button
        className="seven-btn"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="eight-btn"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="nine-btn"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="multiply-btn"
        name="x"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
    <div
      className="div-line3"
    >
      <button
        className="four-btn"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="five-btn"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="six-btn"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="subtract-btn"
        name="-"
        onClick={[Function]}
        type="button"
      >
        −
      </button>
    </div>
    <div
      className="div-line4"
    >
      <button
        className="one-btn"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="two-btn"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="three-btn"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="plus-btn"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="div-line5"
    >
      <button
        className="zero-btn"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="fullstop-btn"
        name="."
        onClick={[Function]}
        type="button"
      >
        ⋅
      </button>
      <button
        className="equals-btn"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
